### DISCUSSION TITLE
Why is there a "like" feature if it seems forbidden?

### DISCUSSION THREAD

**Discussion by RemiFabre** [tweepy/tweepy#1243] [Questions] [answered]
On Twitter's twitter-automation page (https://help.twitter.com/en/rules-and-policies/twitter-automation) it says :
"Automated likes: You may not like Tweets in an automated manner."

Does this mean that we're not allowed to like a Tweet programmatically? Surely they would not let the feature available if it were the case.

Is there a place where "automated manner" is defined? Or is this a gray area?

Thanks in advance,

**Comment by okomarov** [comment]
@RemiFabre As far as I know it's a grey area, you might get suspended then banned.

It also makes sense since you could in principle create an army of bots and push your news agenda, aka fake news, by liking programmatically and creating artificial trends.

**Comment by RemiFabre** [comment]
Do you know if the sentence I quoted has been always there? I'm interested in how it's actually applied.

I'd like to create filters to like specific content so I can manually browse it later on by going to my liked content. I think it should never be more than 100 likes per day in order to be humanly readable.

Do you guys think this would be ok?

**Comment by okomarov** [comment]
I would use collections: https://developer.twitter.com/en/docs/tweets/curate-a-collection/api-reference

**Comment by RemiFabre** [comment]
Thanks for the tip. I would have wanted to let the authors know that their content was of interest to me, the like feature allowed that too.

I'd be interested if anyone has more insight on how this is applied in practice.

**Comment by okomarov** [comment]
You mean you want authors to know that an algorithm liked their content? I’d let users express their interest, if any, off a pre-filtered list.

Otherwise you can end up with a network where bots interact between each other and nobody cares...

Im not criticising your suggestion, it’s just a reflection on the current state of noise versus news.

**Comment by RemiFabre** [comment]
Yes, that would be a risk. But then, they should take the "like" feature out of the api or define the boundaries of its usage.

Would you have an estimate or an article on the "noise vs news ratio"? I read somewhere that in 2018, it was estimated that 15% of the tweets where bot produced.

**Comment by okomarov** [comment]
The seach terms "Twitter bans bots" lands a few results as recent as April. 

To answer your question in the title, Twitter has a like endpoint to let you build additional functionality while keeping the user experience consistent, e.g. TweetDeck.

**Comment by Harmon758** [comment, marked as answer]
> Does this mean that we're not allowed to like a Tweet programmatically? Surely they would not let the feature available if it were the case.

No, there's a difference between automatic and programmatic.
The API includes the functionality because you can programmatically include a method for your product to like a Tweet based on user interaction/input without automatically liking Tweets.

> Is there a place where "automated manner" is defined? Or is this a gray area?

Generally, it means spontaneously and without user interaction/input.

> Do you know if the sentence I quoted has been always there?

The page says it was last updated November 3, 2017.

> they should take the "like" feature out of the api or define the boundaries of its usage

As explained, there are legitimate uses for liking Tweets through the API without doing so automatically, and the limitation for doing so is specifically defined in the rules you linked.

This discussion is not specific or really related to Tweepy and would be better suited on the Twitter Developers Forums. As that help page says:
> If you are a developer and have questions about these Automation Rules, please reach out via https://twittercommunity.com/c/rules-and-policies.

(The link is outdated and should probably be https://twittercommunity.com/c/dev-rules-and-policies.)

A simple search actually comes up with exactly this topic: https://twittercommunity.com/t/automatic-liking-is-not-allowed-but-what-is/71159

**Comment by RemiFabre** [comment]
(Thank you for taking the time and providing quality answers!)
